NEXT-37038 - Remote thumbnail generation docs (#1425)
shopware/docs: guides/plugins/plugins/content/media/remote-thumbnail-generation.md

---
nav:
  title: Remote Thumbnail Generation
  position: 40

---

# Remote Thumbnail Generation

In certain scenarios, you might want to disable the filesystem thumbnail generation in Shopware and use an external CDN service to handle the thumbnails.
This can be beneficial for performance and scalability reasons.
When the remote thumbnail configuration is enabled, the thumbnail images and thumbnail records in the database are not generated by Shopware, but by the external service.

## Configuration

> To use remote thumbnails, you need to adjust the following parameters in your `shopware.yaml`:

Reviewer: ```suggestion
To use remote thumbnails, you need to adjust the following parameters in your `config/packages/shopware.yaml`:
```

> 

Reviewer: Please give the actual yaml reference so they can copy

```yaml
// config/packages/shopware.yaml
shopware:
    media:
        remote_thumbnails:
             enable: true,
             pattern: {mediaUrl}/{mediaPath}?width={width}&ts={mediaUpdatedAt}
```

1. `shopware.media.remote_thumbnails.enable`: Set this parameter to `true` to enable the use of remote thumbnails.

2. `shopware.media.remote_thumbnails.pattern`: This parameter defines the URL pattern for your remote thumbnails. Replace it with your actual URL pattern.

The pattern supports the following variables:

* `mediaUrl`: The base URL of the media file.
* `mediaPath`: The path of the media file relative to the mediaUrl.
* `width`: The width of the thumbnail.
* `height`: The height of the thumbnail.
* `mediaUpdatedAt`: The unix timestamp of the last media change.

> With the pattern set as `{mediaUrl}/{mediaPath}?width={width}&ts={mediaUpdatedAt}`, the resulting URL would be `https://yourshop.example/abc/123/456.jpg?width=80&ts=1718954838`.

Reviewer: ```suggestion
For example, with the pattern set as `{mediaUrl}/{mediaPath}?width={width}&ts={mediaUpdatedAt}`, the thumbnail URL would generated as `https://yourshop.example/abc/123/456.jpg?width=80&ts=1718954838`.
```

---
nav:
  title: Remote Thumbnail Generation
  position: 40

---

> 

Reviewer: ::: note
This feature is available since Shopware 6.6.4.0
:::

# Remote Thumbnail Generation

In certain scenarios, you might want to disable the filesystem thumbnail generation in Shopware and use an external CDN service to handle the thumbnails.
This can be beneficial for performance and scalability reasons.

Starting with version 6.6.4.0, Shopware introduces the feature of remote thumbnail generation.
When the remote thumbnail configuration is enabled, the thumbnail images and thumbnail records in the database are not generated by Shopware, but by the external service.

## Configuration

To use remote thumbnails, you need to adjust the following parameters in your `config/packages/shopware.yaml`:

```yaml

> media:

Reviewer 2: @tamvt can you add the shopware key here as well. Many people just copy the code.

---
nav:
  title: Remote Thumbnail Generation
  position: 40

---

# Remote Thumbnail Generation

::: info
This feature is available starting with Shopware version 6.6.4.0
:::

In certain scenarios, you might want to disable the filesystem thumbnail generation in Shopware and use an external CDN service to handle the thumbnails.
This can be beneficial for performance and scalability reasons.
When the remote thumbnail configuration is enabled, the thumbnail images and thumbnail records in the database are not generated by Shopware, but by the external service.

## Configuration

To use remote thumbnails, you need to adjust the following parameters in your `config/packages/shopware.yaml`:

```yaml
shopware:
  media:
    remote_thumbnails:
      enable: true
      pattern: '{mediaUrl}/{mediaPath}?width={width}&ts={mediaUpdatedAt}'
```

> 1. `shopware.media.remote_thumbnails.enable`: Set this parameter to `true` to enable the use of remote thumbnails.

Reviewer 2: ```suggestion
1. `shopware.media.remote_thumbnails.enable`: Set this parameter to `true` to enable remote thumbnails.
```

2. `shopware.media.remote_thumbnails.pattern`: This parameter defines the URL pattern for your remote thumbnails. Replace it with your actual URL pattern.

The pattern supports the following variables:

* `mediaUrl`: The base URL of the media file.

> * `mediaPath`: The path of the media file relative to the `mediaUrl`.

Reviewer 2: ```suggestion
* `mediaPath`: The media file path relative to the `mediaUrl`.
```

* `width`: The width of the thumbnail.
* `height`: The height of the thumbnail.
* `mediaUpdatedAt`: The timestamp of the last media change.

> For example, by default the pattern was set as `{mediaUrl}/{mediaPath}?width={width}&ts={mediaUpdatedAt}`, the thumbnail URL would be generated as `https://yourshop.example/abc/123/456.jpg?width=80&ts=1718954838`.

Reviewer 2: ```suggestion
For example, by default, the pattern was set as `{mediaUrl}/{mediaPath}?width={width}&ts={mediaUpdatedAt}`, the thumbnail URL would be generated as `https://yourshop.example/abc/123/456.jpg?width=80&ts=1718954838`.
```

## Usage

> Once the configuration is set, Shopware will automatically use the defined pattern to generate URLs for thumbnails.

Reviewer 2: ```suggestion
Once the configuration is set, Shopware will automatically use the defined pattern to generate thumbnail URLs.
```

> These URLs will point to the external CDN service, which should handle the actual generation and delivery of the thumbnail images.

Reviewer 2: ```suggestion
These URLs will point to the external CDN service, which should handle generating and delivering the thumbnail images.
```
